uReplicator-Manager/src/main/java/com/uber/stream/kafka/mirrormaker/manager/core/ControllerHelixManager.java

```diff
@@ -1420,27 +1421,50 @@ private static String getSrc(String pipeline) {
   }
 
   public void disableAutoScaling() {
-    _enableAutoScaling = false;
+    updateClusterConfig(AUTO_SCALING, DISABLE);
   }
 
   public void enableAutoScaling() {
-    _enableAutoScaling = true;
+    updateClusterConfig(AUTO_SCALING, ENABLE);
   }
 
   public boolean isAutoScalingEnabled() {
-    return _enableAutoScaling;
+    HelixConfigScope scope = newClusterConfigScope();
+    Map<String, String> config = _helixAdmin.getConfig(scope, Arrays.asList(AUTO_SCALING));
+    if (config.containsKey(AUTO_SCALING) && config.get(AUTO_SCALING).equalsIgnoreCase(DISABLE)) {
+      return false;
+    }
+    return true;
   }
 
   public void disableAutoBalancing() {
-    _enableRebalance = false;
+    updateClusterConfig(AUTO_BALANCING, DISABLE);
   }
 
   public void enableAutoBalancing() {
-    _enableRebalance = true;
+    updateClusterConfig(AUTO_BALANCING, ENABLE);
   }
 
   public boolean isAutoBalancingEnabled() {
-    return _enableRebalance;
+    HelixConfigScope scope = newClusterConfigScope();
```

Comment:
this code is very similar to `/uReplicator-Controller/src/main/java/com/uber/stream/kafka/mirrormaker/controller/core/HelixMirrorMakerManager.java` 
shall we consider refactor it and put into Utils.

Reply:
ok